src/services/machines/onboardingMachineService.ts

```diff
@@ -121,20 +126,77 @@ export const storePIDCredentials = async (context: Pick<OnboardingMachineContext
 };
 
 export const setupWallet = async (
-  context: Pick<OnboardingMachineContext, 'pinCode' | 'emailAddress' | 'name' | 'biometricsEnabled' | 'pidCredentials' | 'countryCode'>,
+    context: Pick<OnboardingMachineContext, 'pinCode' | 'emailAddress' | 'name' | 'biometricsEnabled' | 'pidCredentials' | 'countryCode'| 'credentialData'>
 ): Promise<WalletSetupServiceResult> => {
   const {pinCode} = context;
   const setup = await Promise.all([
     storagePersistPin({
       value: pinCode,
     }),
+    createSelfIssuedCredential(context)
+        .then((credential) =>
+           agent.ibAddCredentialBranding({
+            vcHash: credential.hash,
+            issuerCorrelationId: credential.issuerCorrelationId,
+            localeBranding: [SphereonWalletIdentityBranding]
+          })
+        ),
```

Comment:
what happens if createSelfIssuedCredential (or storagePersistPin)fails? Can the user go back and try again?
Not needed to do a .catch() block which returns a Promise.reject() ?

Reply:
this would be handled by the machine. as this results in a error and thus the invoke onError gets triggered in the onboarding machine.

not saying this is being handled gracefully as i have not checked, but i would say this is out of scope of this ticket

Reply:
https://sphereon.atlassian.net/browse/MWALL-725